src/components/Monaco/ReactMonacoEditor.tsx

```diff
@@ -36,14 +35,32 @@ export function ReactMonacoEditor({
   ...props
 }: ReactMonacoEditorProps) {
   const theme = useTheme()
+  const [editor, setEditor] = useState<monaco.editor.IStandaloneCodeEditor>()
   const [toolbarState, setToolbarState] = useState<ToolbarState>({
     wordWrap: 'off',
   })
 
+  // Monaco automatically applies word wrap if the content length of a line is >= 10000 characters
+  // In this case, we disable the word wrap button so the internal state is respected
+  const shouldEnableWordWrapButton = () => {
```

Comment:
@e-fisher This was an interesting find. If any line has a content length >= 10000 characters, Monaco overrides the `wordWrap` option. This is likely related to [stopRenderingLineAfter](https://microsoft.github.io/monaco-editor/typedoc/interfaces/editor.IEditorConstructionOptions.html#stopRenderingLineAfter) but changing this option doesn't seem to have any effect on this case specifically other than hiding the rest of the line content.

This was the solution I came up with by disabling the word wrap button and respecting Monaco's internal state. Please let me know what you think.